### PR TITLE
Feature: Generative art toolbar

### LINKED ISSUE #9737: Generative art toolbar
Since there is no way of fullscreen, seeing it on a new tab, and also resetting the animation without reloading the page, there is a drop toolbar 

## design
- updated in small things handoff: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1287-18959&mode=design&t=5apKDpdFtsISv8TI-11
- icons change to k-grey on hover 
- container stays the same, only thing that animates are the icons 
- on desktop it appears when you hover over the img 
- when the reload button is clicked, loader is displayed instead of the item until its possible to click again (use the same we are already using somewhere)

image
image

ignore the gallery item and mobile mockups - it is an older version of the design - the important thing is the button placement 

image

## tooltips 
- it will show small tooltips when hovering over the icon

like this: 
image

image

## Mobile
on touchdevices, there is no hover of the img, so for that case, the toolbar is put underneath the img 

image

- toolbar is then stretched to the full width with icons centered

### PR DESCRIPTION
**Pull request by Temidayo32**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #9737
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [ ] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=14rs1iTduAn9YrqtDwsTZxK9YDbkzDekBstqxc78M6An87fX&usdamount=0)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/koda_eco)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Comment by exezbcz** [comment]
image
image


please use correct font awesome iccons and match the padding (button size)

**Comment by exezbcz** [comment]
even worse
image


check again

**Comment by Temidayo32** [comment]
> even worse image
> 
> check again

@exezbcz  I'm not sure I get the specific element you are referring to here. The reload is working alright on my end here. That is, using the netlify instance generated for this PR. I would appreciate if you can help me know the specific issue here you are pointing out.


Uploading reload - Made with Clipchamp.mp4…

**Comment by exezbcz** [comment]
@Temidayo32 size

**Comment by exezbcz** [comment]
sir, the icons are not centered 
image

**Comment by exezbcz** [comment]
aaand one more thing, the toolbar should show only on generative art

**Comment by prury** [comment]
@Temidayo32 
- Checked again and bar its also showing for non-generative items(should be like exez pointed, only shown on generative art items)
- tooltip also not present

**Comment by Temidayo32** [comment]
> @Temidayo32
> 
> * Checked again and bar its also showing for non-generative items(should be like exez pointed, only shown on generative art items)
> * tooltip also not present

I think I might appreciate some clarification as regards the route for generative and non-generative items. I assumed all generative items are on the `[id]/gallery/` path

**Comment by prury** [comment]
> > @Temidayo32
> > 
> > * Checked again and bar its also showing for non-generative items(should be like exez pointed, only shown on generative art items)
> > * tooltip also not present
> 
> I think I might appreciate some clarification as regards the route for generative and non-generative items. I assumed all generative items are on the `[id]/gallery/` path. It appears to me this is not the case.

cc:
@kodadot/internal 

Also, i think you removed the fullscreen functionality completely, and that should not be the case. 
the bar should only appear to specific genart collections(the ones that are animated)

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -251,6 +268,9 @@ const canPreview = computed(
 
 const activeCarousel = ref(0)
 
+const isLoading = ref(false)
+const generativeArtToolbar = computed(() => route.path.includes('/gallery'))
```
Comment (Jarsen136) [inline components/gallery/GalleryItem.vue:272]: .

**Review by Jarsen136 — CHANGES_REQUESTED** [review]
> I assumed all generative items are on the `[id]/gallery/` path. It appears to me this is not the case.

At least you could check if the `mimeType` is iframe.

**Comment by Temidayo32** [comment]
> > > @Temidayo32
> > > 
> > > * Checked again and bar its also showing for non-generative items(should be like exez pointed, only shown on generative art items)
> > > * tooltip also not present
> > 
> > 
> > I think I might appreciate some clarification as regards the route for generative and non-generative items. I assumed all generative items are on the `[id]/gallery/` path. It appears to me this is not the case.
> 
> cc: @kodadot/internal
> 
> Also, i think you removed the fullscreen functionality completely, and that should not be the case. the bar should only appear to specific genart collections(the ones that are animated)

I don't think the fullscreen functionality is removed

**Comment by vikiival** [comment]
[Closing as this PR used force-push](https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#is-it-of-good-quality)

Please do open a new PR without force push.

Screenshot 2024-03-19 at 11 31 28

**Pull request closed without merging**